src/compiler/utilities.ts

```diff
@@ -1792,9 +1792,10 @@ namespace ts {
         return containerKind === SyntaxKind.InterfaceDeclaration || containerKind === SyntaxKind.TypeLiteral;
     }
 
-    export function isFirstIdentifierOfImplementsClause(node: Node) {
-        return node.parent?.parent?.parent?.kind === SyntaxKind.HeritageClause
-            && (node.parent.parent.parent as HeritageClause).token === SyntaxKind.ImplementsKeyword;
+    export function isFirstIdentifierOfNonEmittingHeritageClause(node: Node): boolean {
+        // Number of parents to climb from identifier is 2 for `implements I`, 3 for `implements x.I`

```

Comment:
> 3 for `implements x.I`

This is namespaced, right? In the case of a PropertyAccessExpression, we only call this function on the left-most identifier, so it doesn’t matter how many dots are in the dotted name; it will still be 3 levels from the left-most identifier to the HeritageClause (`HeritageClause > ExpressionWithTypeArguments > PropertyAccessExpression > Identifier`). In the lone identifier case, it’s 2 (`HeritageClause > ExpressionWithTypeArguments > Identifier`).

Reply:
:+1: I forgot about that.